proposals/csharp-12.0/collection-expressions.md

```diff
@@ -210,6 +210,51 @@ ImmutableArray<int> ia =
     ImmutableArray.Create((ReadOnlySpan<int>)__tmp);
 ```
 
+## Spreads
+[spreads]: #spreads
+
+A *spread element* enumerates the elements of a collection and adds those elements, in order, to the containing *collection expression*.
+A spread element is only valid as an element of a collection expression.
+
+```diff
+spread_element
+  : '..' expression
+  ;
+```
+
+Compile-time processing of the spread element relies on the *collection type*, *enumerator type*, and *iteration type* of the *expression*, as those types are defined for the [*foreach statement*](https://github.com/dotnet/csharpstandard/blob/standard-v7/standard/statements.md#1395-the-foreach-statement).
+
+If the spread element expression is an *inline array type* with element type `T`, the collection type is `System.ReadOnlySpan<T>`, the enumerator type is the return type of `System.ReadOnlySpan<T>.GetEnumerator()`, and the iteration type is `T`.
+
+If the collection type, enumerator type, or iteration type cannot be determined for the spread element expression, an error is reported.
+
+For a spread element `..x`, with collection type `C`, enumerator type `E`, and iteration type `T`, `ref T`, or `ref readonly T`, the spread element is essentially expanded to:
+```csharp
+E __e = ((C)(x)).GetEnumerator();
+try
+{
+    while (__e.MoveNext())
+    {
+      T __t = (T)__e.Current;
+      // ... add __t to containing collection
+    }
+}
+finally
+{
+    // ... dispose __e
+}
+```
+
+The compiler may use an alternate approach for enumerating and adding the elements of the spread if the behavior is otherwise consistent.
+
+The `finally` block for disposing the enumerator matches the [*foreach statement*](https://github.com/dotnet/csharpstandard/blob/standard-v7/standard/statements.md#1395-the-foreach-statement).
+
+The order for enumerating elements of arrays matches the [*foreach statement*](https://github.com/dotnet/csharpstandard/blob/standard-v7/standard/statements.md#1395-the-foreach-statement).
+
+The `GetEnumerator` method identified in the compile-time processing may be an extension method.
+
+The `Current` property identified in the compile-time processing may be *ref returning*.
```

Comment:
do we need language talking about when teh expression that is being spread is evaluated, versus when teh values are enumerated, and what guarantees there are (or are not) around that?  (or does that already exist?

Reply:
The order of evaluation should be covered in the Construction section that follows this one.